interactors/addressNonceHandler_test.go

```diff
@@ -0,0 +1,378 @@
+				return &data.Account{Nonce: blockchainNonce}, nil
+			},
+		}
+		anh, _ := NewAddressNonceHandler(proxy, testAddress)
+		txArgs := createTxArgs()
+		tx := createTx(txArgs.GasPrice, txArgs)
+		_, err := anh.SendTransaction(context.Background(), tx)
+		require.Nil(t, err)
+		require.Equal(t, 1, len(anh.transactions))
+
+		anh.computedNonce = blockchainNonce
+		anh.lowestNonce = 80
+		err = anh.ReSendTransactionsIfRequired(context.Background())
+		require.Equal(t, anh.computedNonce, anh.lowestNonce)
+		require.Equal(t, 0, len(anh.transactions))
+		require.Nil(t, err)
+	})
+	t.Run("len(anh.transactions) == 0", func(t *testing.T) {
+		anh, _ := NewAddressNonceHandler(&testsCommon.ProxyStub{}, testAddress)
+		txArgs := createTxArgs()
+		tx := createTx(txArgs.GasPrice, txArgs)
+		_, err := anh.SendTransaction(context.Background(), tx)
+		require.Nil(t, err)
+		require.Equal(t, 1, len(anh.transactions))
+
+		anh.computedNonce = 100
+		anh.lowestNonce = 80
+		err = anh.ReSendTransactionsIfRequired(context.Background())
+		require.Equal(t, anh.computedNonce, anh.lowestNonce)
+		require.Equal(t, 0, len(anh.transactions))
+		require.Nil(t, err)
+	})
+	t.Run("lowestNonce should be recalculated each time", func(t *testing.T) {
+		blockchainNonce := uint64(100)
+		proxy := &testsCommon.ProxyStub{
+			GetAccountCalled: func(address core.AddressHandler) (*data.Account, error) {
+				return &data.Account{Nonce: blockchainNonce - 1}, nil
+			},
+		}
+		anh, _ := NewAddressNonceHandler(proxy, testAddress)
+		txArgs := createTxArgs()
+		tx := createTx(txArgs.GasPrice, txArgs)
+		tx.Nonce = blockchainNonce + 1
+		_, err := anh.SendTransaction(context.Background(), tx)
+		require.Nil(t, err)
+		require.Equal(t, 1, len(anh.transactions))
+
+		anh.computedNonce = blockchainNonce + 2
+		anh.lowestNonce = blockchainNonce
+		err = anh.ReSendTransactionsIfRequired(context.Background())
+		require.Equal(t, blockchainNonce+1, anh.lowestNonce)
+		require.Equal(t, 1, len(anh.transactions))
+		require.Nil(t, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		blockchainNonce := uint64(100)
+		proxy := &testsCommon.ProxyStub{
+			GetAccountCalled: func(address core.AddressHandler) (*data.Account, error) {
+				return &data.Account{Nonce: blockchainNonce - 1}, nil
+			},
+			SendTransactionsCalled: func(txs []*data.Transaction) ([]string, error) {
+				return make([]string, 0), nil
+			},
+		}
+		anh, _ := NewAddressNonceHandler(proxy, testAddress)
+		txArgs := createTxArgs()
+		tx := createTx(txArgs.GasPrice, txArgs)
+		tx.Nonce = blockchainNonce
+		_, err := anh.SendTransaction(context.Background(), tx)
+		require.Nil(t, err)
+		require.Equal(t, 1, len(anh.transactions))
+
+		anh.computedNonce = blockchainNonce
+
+		err = anh.ReSendTransactionsIfRequired(context.Background())
+		require.Equal(t, 1, len(anh.transactions))
+		require.Nil(t, err)
+	})
+}
+
+func TestAddressNonceHandler_fetchGasPriceIfRequired(t *testing.T) {
+	t.Parallel()
```

Comment:
this one can be moved inside t.Run

Reply:
I am not sure about that. Moving it inside t.Run wouldn\`t make `TestAddressNonceHandler_fetchGasPriceIfRequired` be executed only after the other `Test...` function will be executed, and not in parallel ?

Reply:
as discussed, we`ll make the t.Run just a simple comment

Reply:
ok.. or could have updated the `TestAddressNonceHandler_fetchGasPriceIfRequired` func name